bookkeeper-common/src/main/java/org/apache/bookkeeper/common/checksum/Checksums.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.bookkeeper.common.checksum;
+
+import java.util.zip.CRC32;
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+import org.apache.bookkeeper.common.util.Java;
+
+/**
+ * A util classes for providing accesses to different {@link ChecksumFactory}s.
+ */
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public final class Checksums {
+
+    /**
+     * Returns a {@link CRC32} factory.
+     *
+     * @return a {@link CRC32} factory.
+     */
+    public static ChecksumFactory crc32() {
+        return () -> new CRC32Checksum();
+    }
+
+    /**
+     * Returns a default <tt>CRC32C</tt> factory.
+     *
+     * @return a default <tt>CRC32C</tt> factory.
+     */
+    public static ChecksumFactory crc32c() {
+        if (Java.IS_JAVA9_COMPATIBLE) {
```

Comment:
I think it is better to use 'feature detection' than checking for java version. I mean to check if the needed classes are available at runtime.

Using versions is fragile, and things may break easily, see all of the problems with commons-lang.